### PR TITLE
support define-syntax-parser builtin

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

File: syntaxes/racket.tmLanguage.json [hunk 1]
```diff
@@ -166,7 +166,7 @@
...
...random-generator! | vector->values | vector-append | vector-argmax | vector-argmin | vector-cas! | vector-copy | vector-copy! | vector-count | vector-drop | vector-drop-right | vector-fill! | vector-filter | vector-filter-not | vector-immutable | vector-length | vector-map | vector-map! | vector-member | vector-memq | vector-memv | vector-ref | vector-set! | vector-set\\*! | vector-set-performance-stats! | vector-split-at | vector-split-at-right | vector-take | vector-take-right | vector\\? | version | void | void\\? | weak-box-value | weak-box\\? | weak-set | weak-seteq | weak-seteqv | will-execute | will-executor\\? | will-register | will-try-execute | with-input-from-bytes | with-input-from-string | with-output-to-bytes | with-output-to-string | would-be-future | wrap-evt | wrapped-extra-arg-arrow-extra-neg-party-argument | wrapped-extra-arg-arrow-real-func | wrapped-extra-arg-arrow\\? | writable<%> | write | write-byte | write-bytes | write-bytes-avail | write-bytes-avail\\* | write-bytes-avail-evt | write-bytes-avail\\/enable-break | write-char | write-special | write-special-avail\\* | write-special-evt | write-string | writeln | xor | zero\\?\n) (?=$|[()\\[\\]{}\",'`;\\s])",
```
Comment (dannypsnl) [inline syntaxes/racket.tmLanguage.json:169]: Shouldn't we expand this line? It's really hard to find out what's changed on GitHub

Comment (Eugleo) [reply syntaxes/racket.tmLanguage.json:169]: You surely can.

**Comment by Eugleo** [comment]
I'm not sure whether it makes sense, but in the past I generated the `tmLanguage.json` file from the `.yaml` file that's in `syntaxes/src`. We should discuss whether it makes sense to use the yaml file, because if it does, it would be better to make the changes there and then just generate the json file from it to keep them both in sync. What do you think?

**Comment by dannypsnl** [comment]
Yes, that also confuse me once I find out there has a YAML. I think if keep YAML, maybe should ignore JSON, and generate JSON on deployment. After some read, I think YAML looks easier for maintaining.

**Comment by dannypsnl** [comment]
And I also found some interesting:
圖片

there have duplicated `define-syntax`...

**Comment by Eugleo** [comment]
Good point. Would you mind removing these duplicated values, and setting up the GitHub action to generate the json file automatically?

**Comment by dannypsnl** [comment]
Sure, whats your command for yaml to json?

**Comment by Eugleo** [comment]
Tbh I did all of this such a long time ago that I have no idea. We will need to find something suitable.

**Comment by dannypsnl** [comment]
Ok, maybe I also do some research about conversion tool.

**Comment by shocoman** [comment]
Actually, there about 1500 duplicates in that list 
image

**Comment by dannypsnl** [comment]
@Eugleo I finally find one, I think you might use https://github.com/pedro-w/vscode-tmlanguage

This plugin has a command `Convert to tmLanguage JSON` can be triggered via `shift + command + p` in vscode.

**Comment by dannypsnl** [comment]
1. I want to rebuild YAML file from JSON since the current YAML file is actually invalid.
2. Then merge these branch first for new keywords support(`define-syntax-parser`).
3. And we have another PR for remove duplicates

**Pull request closed without merging**